File: meme_inator_back/apps/profiles/application/usecases/update_profile_counters_usecase.py
from typing import Dict
from uuid import UUID
from apps.profiles.domain.entities.profile_entity import ProfileEntity
from apps.profiles.domain.usecases.iupdate_profile_counters_usecase import IUpdateProfileCountersUsecase
from apps.profiles.domain.irepositories.i_profile_repository import IProfileRepository
from core.results import Result

class UpdateProfileCountersUsecase(IUpdateProfileCountersUsecase):
    def __init__(self, profile_repo: IProfileRepository):
        self.profile_repo = profile_repo

    def execute(self, user_id: UUID, increments: Dict[str, int]) -> Result[ProfileEntity]:
        return self.profile_repo.update_counters(user_id, increments)

File: meme_inator_front/lib/features/profiles/domain/usecases/update_profile_counters_usecase.dart
import 'package:meme_inator_front/core/iusecase.dart';
import 'package:meme_inator_front/core/results.dart';
import 'package:meme_inator_front/features/profiles/domain/entities/profile_entity.dart';
import 'package:meme_inator_front/features/profiles/domain/interfaces/iprofile_repository.dart';
import 'package:uuid/uuid.dart';

class UpdateProfileCountersInput {
  final Uuid userId;
  final Map<String, int> increments;

  const UpdateProfileCountersInput({
    required this.userId,
    required this.increments,
  });
}

class UpdateProfileCountersUsecase
    extends IUseCase<UpdateProfileCountersInput, ProfileEntity> {
  final IProfileRepository profileRepo;

  UpdateProfileCountersUsecase(this.profileRepo);

  @override
  Future<Result<ProfileEntity>> execute({
    required UpdateProfileCountersInput valObj,
  }) async {
    throw UnimplementedError('UpdateProfileCountersUsecase.execute not implemented');
  }
}
